pkg/apis/flowcontrol/types.go

```diff
@@ -418,9 +418,11 @@ type LimitedPriorityLevelConfiguration struct {
 	//
 	// Bigger numbers mean a larger nominal concurrency limit,
 	// at the expense of every other priority level.
-	// This field has a default value of 30.
+	//
+	// If not specified, this field defaults to a value of 30.
+	//
 	// +optional
-	NominalConcurrencyShares int32
+	NominalConcurrencyShares *int32
```

Comment:
we shouldn't need to change the internal type to a pointer, right?

v1beta3:
* `0` value defaults to 30 IFF there's no annotation set
* `0` value stays 0 IFF there's the annotation set
* whether it defaults to 30 or stays at 0, we have a value to convert to the non-pointer internal type

v1:
* nil value defaults to 30
* 0 stays 0
* whether it defaults to 30 or stays at 0, we have a value to convert to the non-pointer internal type

Reply:
I think we could use either representation approach in the internal type. Is there a problem with using the pointer-to-int?

Reply:
it's a change, which makes all internal code have to change to dereference... if all external types will apply defaults, we don't need to round-trip "absent", so I'd tend to just leave the internal type and code alone

Reply:
@liggitt : thanks, I see your point. To be clear, you are saying that the internal type can have a non-pointer type for NominalConcurrencyShares while also NOT ever having the special annotation that gets used in v1beta3, right? I would be OK with that.

Reply:
> we don't need to round-trip "absent",

okay, i thought I needed to preserve "absent" in the internal type for future, but I agree keeping the internal non-pointer makes this much cleaner.